### PR TITLE
Regression in matrix docs

### PR DESCRIPTION
**Pull request by sylee957**
#### Brief description of what is fixed or changed

I have found a lot of matrix docs for methods like `det()` `eigenvals()` missing after changes made between [1.0](https://docs.sympy.org/1.0/modules/matrices/matrices.html#matrixbase-class-reference) and [1.1](https://docs.sympy.org/1.1/modules/matrices/matrices.html#matrixbase-class-reference).

I think that there had been several changes to separate matrix classes,
however docs had not been updated since then.
I think sphinx does not automatically load parent classes of `MatrixBase`
So I have added all the matrix classes to the sphinx to restore the docs.

I'm still working on the `MatrixReduction`, because there are some sphinx formatting errors making my build to fail.
And I would also want to hear your opinions about where the classes like `MatrixDeprecated` and `DeferredVector` should be added to.

#### Release Notes

*  matrices
   * Added missing docs from matrices.py

### CODE REVIEW

**Comment by jksuom** [comment]
It looks like `MatrixDeprecated` need not be documented. (Its methods should not be used in new code.)
I am not sure about how much `DeferredVector` is used. Can it be replaced by other types of matrices?

**Comment by sylee957** [comment]
@jksuom 
I think that `DeferredVector` is used only in `lambdify`, and I don't think it is closely related to any `Matrix` operations.
Maybe it can be added to a separate page `deferredvector.rst`, but still within matrices section, so that any future attempt to move the class outside of the module, or deprecating can be notified about changes to be made in docs.

I also think `MatrixDeprecated` class can also be treated as in same fashion as above,
because the community is not totally having consensus on making a hard deprecation,
and accessing all those methods still does not throw any deprecation warnings yet.

From [Numpy's](https://docs.scipy.org/doc/numpy/reference/generated/numpy.random.random_integers.html?highlight=deprecated) example of treating redundancy,
I think they can still be documented in, if there is a clear mentioning that it is redundant.
If some docs are getting completely gone one day, I think it can also confuse users who are trying to search references for the features they were using.

**Comment by sidhantnagpal** [comment]
Restarted the timed out job.

File: sympy/matrices/matrices.py [hunk 1]
```diff
@@ -645,7 +645,7 @@ def _row_reduce(self, iszerofunc, simpfunc, normalize_last=True,
         normalize : whether pivot rows should be normalized so that
             the pivot value is 1
         zero_above : whether entries above the pivot should be zeroed.
-            If `zero_above=False`, an echelon matrix will be returned.
+            If `zero_above=False` , an echelon matrix will be returned.
```
Comment (jksuom) [inline sympy/matrices/matrices.py]: What is the purpose of extra space?

File: sympy/matrices/matrices.py [hunk 2]
```diff
@@ -992,14 +992,14 @@ def rowspace(self, simplify=False):
...
-        to vectors supplied in `vecs`.
+        to vectors supplied in `vecs` .
 
         Arguments
         =========
 
         vecs : vectors to be made orthogonal
-        normalize : bool. Whether the returned vectors
-                    should be renormalized to be unit vectors.
+        normalize : bool. Whether the returned vectors should be renormalized
```
Comment (jksuom) [inline sympy/matrices/matrices.py]: I would recommend something like this:

    normalize : bool
        If true, return an orthonormal basis.

**Comment by sylee957** [comment]
I could confirm that there is no deed of extra spacing before comma, but I think this spacing is necessary
```By default SymPy's `simplify` is used.```

And, I had also changed `Arguments` to `Parameters` because sphinx does not render `Arguments` as some tabular form, but rather as plane paragraph.

Can anyone review? @jksuom @sidhantnagpal

File: sympy/matrices/matrices.py [hunk 3]
```diff
@@ -994,12 +994,15 @@ def orthogonalize(cls, *vecs, **kwargs):
...
 
-        Arguments
-        =========
+        Parameters
+        ==========
+
+        vecs :
+            vectors to be made orthogonal
 
-        vecs : vectors to be made orthogonal
```
Comment (sidhantnagpal) [inline sympy/matrices/matrices.py]: This could be left as-is (without a line break) and for `normalize` you could address the previous comments.

Comment (sylee957) [reply sympy/matrices/matrices.py]: @sidhantnagpal 
I think there is a clear difference between how docs render with and without line break,
but is first one preferable?
image

image

Comment (sidhantnagpal) [reply sympy/matrices/matrices.py]: A newline between the two parameters `vecs` and `normalize` will make it clear I suppose.

Comment (sylee957) [reply sympy/matrices/matrices.py]: @sidhantnagpal 
Actually, I don't think that any additional newlines are recognized as-is.
I think the docs are parsed in some way, and only some formats are being accepted.

I think sympy uses numpy's formatting sheet, as you can see some copies of `docscrape.py`, `numpydoc.py` in `sympy/doc/ext`
And from the numpy's doc tutorial
https://numpydoc.readthedocs.io/en/latest/format.html#sections
I think either
```
param : type
    description
```
or
```
param :
    description
```
```
param : type
```
are correct forms.

File: sympy/matrices/matrices.py [hunk 4]
```diff
@@ -853,7 +853,7 @@ def rref(self, iszerofunc=_iszero, simplify=False, pivots=True, normalize_last=T
             act as a pivot.  `lambda x: x.is_zero` is used by default.
         simplify : Function
             A function used to simplify elements when looking for a pivot.
-            By default SymPy's `simplify`is used.
+            By default SymPy's `simplify` is used.
```
Comment (sidhantnagpal) [inline sympy/matrices/matrices.py]: Single back-ticks are used for LaTeX. You should use double back-ticks for code.

File: sympy/matrices/matrices.py (hunk 3 above)
Comment (sidhantnagpal) [reply sympy/matrices/matrices.py]: I think you might have to drop the colon as well if that is the case. See https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html#example-numpy.

> ```
> name : type
>     description
> 
>     The description may span multiple lines. Following lines
>     should be indented to match the first line of the description.
>     The ": type" is optional.
> 
>     Multiple paragraphs are supported in parameter
>     descriptions.
> ```

**Comment by sylee957** [comment]
@sidhantnagpal 

I think there are more critical cases of the misuse of inline interpreted texts, for displaying snakecases though.
So I have changed every cases from `rref` which were using them, but makes no sense of any mathematical notation.

image

I am aware of a lot of badly formatted docs all around the matrices module, which had been masked for more than one year.
It can be a lot of work though.
And I had originally come up with an idea here, of doing just some minimal change to make the docs explicit. And probably opening up some separate thread regarding the issues.

**Comment by sidhantnagpal** [comment]
This is a known issue. See #13519.

> doing just some minimal changes to make the docs explicit

That should be fine for now.

**Comment by smichr** [comment]
I don't see a wiki page giving a docstring formatting guide (like when to use `r'''`; when to use `foo` or ``foo``; Examples needs a line of `=` after it and then a blank line but `See Also` doesn't need the blank line, etc...). Even if that page pointed to another pages (like on numpy) it would be helpful.

**Pull request merged**